### ISSUE TITLE
crash when listing routes

### ISSUE THREAD

**Issue by spring2maz** [emqx/emqx#1812]
#### Environment

- OS: alpine
- Erlang/OTP: 21
- EMQ: v3.0-beta.2

#### Description

```
/opt/emqx $ ./bin/emqx_ctl routes list
RPC to fe76e6f734d9@172.17.0.3 error: {case_clause,
                                       [{route,<<"some/#">>,
                                         'b37a112d3591@172.17.0.4'},
                                        {route,<<"some/#">>,
                                         {<<"1">>,
                                          'b37a112d3591@172.17.0.4'}}]}
```
steps to reproduce:
- start two containers emq1, emq2 and join cluster.
- start two subscribers, one subscribe to `some/#` the other subscribe to `$share/1/some/#`, both connects to emq1.
- execute command `./bin/emqx_ctl routes list` from emq2

crash log:
```
18:14:29.772 [error] [CTL] CMD Error:{case_clause,[{route,<<"some/#">>,'b37a112d3591@172.17.0.4'},{route,<<"some/#">>,{<<"1">>,'b37a112d3591@172.17.0.4'}}]}, Stacktrace:[{emqx_mgmt_cli,dump,2,[{file,"src/emqx_mgmt_cli.erl"},{line,630}]},{emqx_ctl,run_command,2,[{file,"src/emqx_ctl.erl"},{line,71}]},{rpc,'-handle_call_call/6-fun-0-',5,[{file,"rpc.erl"},{line,197}]}]
```

**Comment by spring2maz** [comment]
crash log:
```
18:14:29.772 [error] [CTL] CMD Error:{case_clause,[{route,<<"some/#">>,'b37a112d3591@172.17.0.4'},{route,<<"some/#">>,{<<"1">>,'b37a112d3591@172.17.0.4'}}]}, Stacktrace:[{emqx_mgmt_cli,dump,2,[{file,"src/emqx_mgmt_cli.erl"},{line,630}]},{emqx_ctl,run_command,2,[{file,"src/emqx_ctl.erl"},{line,71}]},{rpc,'-handle_call_call/6-fun-0-',5,[{file,"rpc.erl"},{line,197}]}]```

**Comment by turtleDeng** [comment]
@spring2maz The issue has been fixed and will be released at 3.0-beta.3. https://github.com/emqx/emqx-management/commit/04e87aa53359d9742f1023b4f13b7b080587f73a

**Comment by phanimahesh** [comment]
@turtleDeng I saw the same RPC error in emqx v3.2.0 when listing routes. Have not isolated the cause yet.
